bin/decode_token.sh

```diff
@@ -9,11 +9,16 @@
 decode_token() {
     #
     # token is 3 base64 segments separated by dots
-    # we want the middle one.
-    # the ending of the base64 segment is sometimes
-    # chopped, so ignore errors about that
+    # we want the middle one
     #
-    sed -e 's/.*\.\(.*\)\..*/\1==/' "$1" 2>/dev/null | base64 -d  2>/dev/null
+    sed -e 's/.*\.\(.*\)\..*/\1==/' "$1" | base64 -d 2>/dev/null
+}
+
+pretty_print() {
+   # sed regexps:
+   #  - break line and indent after (some) commas,
+   #  - put braces on their own line
+   sed -e 's/:[^:]*[]"0-9],/&\n  /g' -e 's/[{}]/\n&\n  /g'
```

Comment:
Pardon my lack of `sed` knowledge, but in the first substitution, is there an extra `]` in the search part?  Or is it missing an opening `[` before `0-9`?  Or am I just wrong?

Also, I assume that you're adding the pretty_print() function because you don't want us to rely on the installation of `jq`, right?